src/main/java/com/autotune/analyzer/recommendations/engine/RecommendationEngine.java

```diff
@@ -1984,213 +1979,255 @@ public void fetchMetricsBasedOnProfileAndDatasource(KruizeObject kruizeObject, T
+                    namespaceData = new NamespaceData();
+                    namespaceData.setNamespace_name(namespace);
+                    k8sObject.setNamespaceData(namespaceData);
+                }
 
-                            String dateMetricsUrl = String.format(KruizeConstants.DataSourceConstants.DATE_ENDPOINT_WITH_QUERY,
-                                    dataSourceInfo.getUrl(),
-                                    URLEncoder.encode(queryToEncode, CHARACTER_ENCODING)
-                            );
-                            LOGGER.info(dateMetricsUrl);
-                            client.setBaseURL(dateMetricsUrl);
-                            JSONObject genericJsonObject = client.fetchMetricsJson(KruizeConstants.APIMessages.GET, "");
-                            JsonObject jsonObject = new Gson().fromJson(genericJsonObject.toString(), JsonObject.class);
-                            JsonArray resultArray = jsonObject.getAsJsonObject(KruizeConstants.JSONKeys.DATA).getAsJsonArray(KruizeConstants.DataSourceConstants.DataSourceQueryJSONKeys.RESULT);
-                            // Process fetched metrics
-                            if (null != resultArray && !resultArray.isEmpty()) {
-                                resultArray = resultArray.get(0)
-                                        .getAsJsonObject().getAsJsonArray(KruizeConstants.DataSourceConstants.DataSourceQueryJSONKeys.VALUE);
-                                long epochTime = resultArray.get(0).getAsLong();
-                                String timestamp = sdf.format(new Date(epochTime * KruizeConstants.TimeConv.NO_OF_MSECS_IN_SEC));
-                                Date date = sdf.parse(timestamp);
-                                Timestamp dateTS = new Timestamp(date.getTime());
-                                interval_end_time_epoc = dateTS.getTime() / KruizeConstants.TimeConv.NO_OF_MSECS_IN_SEC
-                                        - ((long) dateTS.getTimezoneOffset() * KruizeConstants.TimeConv.NO_OF_SECONDS_PER_MINUTE);
-                                int maxDay = Terms.getMaxDays(kruizeObject.getTerms());
-                                LOGGER.info(KruizeConstants.APIMessages.MAX_DAY, maxDay);
-                                Timestamp startDateTS = Timestamp.valueOf(Objects.requireNonNull(dateTS).toLocalDateTime().minusDays(maxDay));
-                                interval_start_time_epoc = startDateTS.getTime() / KruizeConstants.TimeConv.NO_OF_MSECS_IN_SEC
-                                        - ((long) startDateTS.getTimezoneOffset() * KruizeConstants.TimeConv.NO_OF_MSECS_IN_SEC);
+                List<Metric> metricList = metricProfile.getSloInfo().getFunctionVariables();
+
+                // Iterate over metrics and aggregation functions
+                for (Metric metricEntry : metricList) {
+                    if (metricEntry.getName().startsWith(AnalyzerConstants.NAMESPACE) && !metricEntry.getName().equals("namespaceMaxDate")) {
```

Comment:
Instead of having this check create a sublist from metricList based on the conditions you've provided, you can use Java Streams to filter
```

list<Metric> nameSpaceMetrics = metricList.stream()
            .filter(metricEntry -> metricEntry.getName().startsWith(AnalyzerConstants.NAMESPACE)
                    && !metricEntry.getName().equals("namespaceMaxDate"))
            .collect(Collectors.toList());
```

and you can avoid above if check

Reply:
Updated.